fix T1327096 (#268)
DevExpress/dotnet-eud: eud-for-devexpress-reports/reporting-for-desktop/report-designer/report-designer-for-winforms/create-reports/master-detail-reports-with-detail-report-bands.md

---
title: Master-Detail Reports with Detail Report Bands
author: Anna Gubareva
---
# Master-Detail Reports with Detail Report Bands

> This tutorial illustrates how to display hierarchical data in a master-detail report using nested [Detail Report bands](../introduction-to-banded-reports.md). This approach is effective if your data source contains master-detail relationship. Another way is described at [Master-Detail Reports with Subreports](master-detail-reports-with-subreports.md).

Reviewer: ```suggestion
This tutorial shows how to display hierarchical data in a master-detail report with nested [Detail Report bands](../introduction-to-banded-reports.md). Use this approach when the data source contains a master-detail relationship. Another approach is described in [Master-Detail Reports with Subreports](master-detail-reports-with-subreports.md).
```

1. [Create a new report](../add-new-reports.md) or [open an existing one](../open-reports.md).

2. [Bind the report](../bind-to-data.md) to a required data source and provide it with a master-detail relationship as described in the [Bind a Report to a Database](../bind-to-data/bind-a-report-to-a-database.md) topic.

3. Drop the required data fields from the [Field List](../report-designer-tools/ui-panels/field-list.md) onto the [Detail](../introduction-to-banded-reports.md) band.

    ![](../../../images/eurd-win-master-detail-drop-fields-for-master-layout.png)

> 4. Create a [Detail Report Band](../introduction-to-banded-reports.md) by right-clicking the report's surface. In the invoked context menu, select **Insert Detail Report**, and then, select the master-detail relationship's name.

Reviewer: ```suggestion
4. Create a [Detail Report Band](../introduction-to-banded-reports.md) by right-clicking the report's surface. In the context menu, select **Insert Detail Report**, and then select the name of the master-detail relationship.
```

    ![](../../../images/eurd-win-master-detail-insert-detail-report-band.png)

>     This sets the detail report's **Data Source** and **Data Member** properties automatically.

Reviewer: Sets them to what?

Author: ```suggestion
    This sets the detail report’s **DataSource** and **DataMember** properties to the selected master-detail relationship automatically.
```

    ![](../../../images/eurd-win-master-detail-data-member-property.png)

> 5. Switch to the **Field List**, select the data fields while holding down CTRL or SHIFT and drag-and-drop them onto the Detail band.

Reviewer: ```suggestion
5. Switch to the **Field List**, select the data fields while holding CTRL or SHIFT, and drag them to the Detail band.
```

    ![](../../../images/eurd-win-master-detail-drop-fields-for-detail-layout.png)

>     Drag fields from the data category that matches the current detail level (the table specified in the most inner report’s DataMember). This ensures that each detail record is displayed correctly.

Reviewer: ```suggestion
    Drag fields from the data category that matches the current detail level, that is, the table specified in the innermost report’s DataMember. This ensures that each detail record is displayed correctly.
```

>     You can display parent values in a detail row using expressions (for example, [Categories.CategoryName]). If a field is not found at the current level, the expression engine automatically resolves it from the parent level.

Reviewer: ```suggestion
    You can display parent values in a detail row using expressions, for example, [Categories.CategoryName]. If a field is not found at the current level, the expression engine automatically resolves it from the parent level.
```
